### DISCUSSION TITLE
Controlling initially focused element in modal dialog

### DISCUSSION THREAD

**Discussion by richbachman** [adobe/react-spectrum#4373] [Q&A]
I'm looking at the new `react-aria-components`, and am excited about using them for an upcoming project.

Is it possible to control what element inside of a modal dialog is initially focused? Maybe something to how Reach UI handles it with a prop? https://reach.tech/dialog/#dialog-initialfocusref

I see in [this example](https://react-spectrum.adobe.com/react-aria/Dialog.html#example) the `autofocus` prop passed to `TextField`, but what if I want to set the focus to a specific button?

Thank you in advance for the help.

**Comment by devongovett** [comment]
The `Button` component also accepts an `autoFocus` prop so you could use that. Alternatively, if you need to focus an element that doesn't support `autoFocus` for some reason, you could make a ref and focus it in a `useLayoutEffect` when the dialog mounts.
